### PR TITLE
Fix/msvc issue

### PR DESCRIPTION
**Pull request by granger35**
Replace C++ array by a std::vector to avoid delete.

### CODE REVIEW

File: SolARService_MappingAndRelocalizationProxy/src/RelocalizationAndMappingGrpcServiceImpl.cpp [hunk 1]
```diff
@@ -583,8 +583,7 @@ RelocalizationAndMappingGrpcServiceImpl::buildSolARImage(const Frame* frame,
             {
                 // Convert to CV_8UC3 because otherwise convertToSolar() will fail
                 const char* image_buffer = frame->image().data().c_str();
-                long destBufferSize = frame->image().data().size() - (frame->image().data().size() / 4);
-                char bgr_buffer[destBufferSize];
+                std::vector<char> bgr_buffer;
                 for (long j = 0, k = 0; j < frame->image().data().size(); j += 4, k += 3)
                 {
                   bgr_buffer[k] = image_buffer[j];
```
Comment (jim-bcom) [inline SolARService_MappingAndRelocalizationProxy/src/RelocalizationAndMappingGrpcServiceImpl.cpp]: The [documentation](https://en.cppreference.com/w/cpp/container/vector/operator_at) for `operator[]` states that `Accessing a nonexistent element through this operator is undefined behavior.`. 

I think the vector must either be contructed with the previously computed size (destBufferSize) using ctor (4), in order to use `[]`, or keep declaration as it is, and use the `push_back()` method instead to add elements, which removes the need for the `k` variable.

File: SolARService_MappingAndRelocalizationProxy/src/RelocalizationAndMappingGrpcServiceImpl.cpp [hunk 2]
```diff
@@ -615,8 +614,7 @@ RelocalizationAndMappingGrpcServiceImpl::buildSolARImage(const Frame* frame,
 
                 // Convert to CV_8UC3 because otherwise convertToSolar() will fail
                 char* image_buffer = (char*)temp_image->data();
-                long destBufferSize = temp_image->getBufferSize() - (temp_image->getBufferSize() / 4);
-                char bgr_buffer[destBufferSize];
+                std::vector<char> bgr_buffer;
```
Comment (jim-bcom) [inline SolARService_MappingAndRelocalizationProxy/src/RelocalizationAndMappingGrpcServiceImpl.cpp:617]: Ditto

File: SolARService_MappingAndRelocalizationProxy/src/SolARService_MappingAndRelocalizationProxy.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@
 #include <iostream>
 #include <signal.h>
 
-#include <unistd.h>
+//#include <unistd.h>
```
Comment (jim-bcom) [inline SolARService_MappingAndRelocalizationProxy/src/SolARService_MappingAndRelocalizationProxy.cpp]: Remove completely if not useful?

File: packagedependencies.txt [hunk 4]
```diff
@@ -1,7 +1,2 @@
-SolARModuleOpenCV|0.11.0|SolARModuleOpenCV|SolARBuild@github|https://github.com/SolarFramework/SolarModuleOpenCV/releases/download
-SolARModuleOpenGL|0.11.0|SolARModuleOpenGL|SolARBuild@github|https://github.com/SolarFramework/SolarModuleOpenGL/releases/download
-SolARModuleTools|0.11.0|SolARModuleTools|SolARBuild@github|https://github.com/SolarFramework/SolARModuleTools/releases/download
-SolARModuleFBOW|0.11.0|SolARModuleFBOW|SolARBuild@github|https://github.com/SolarFramework/SolARModuleFBOW/releases/download
-SolARPipelineRelocalization|0.11.0|SolARPipelineRelocalization|SolARBuild@github|https://github.com/SolarFramework/Sample-Relocalization/releases/download
-SolARPipelineMappingAndRelocalizationFrontend|0.11.0|SolARPipelineMappingAndRelocalizationFrontend|SolARBuild@github|https://github.com/SolarFramework/Sample-Relocalization/releases/download
-
+SolARFramework|0.11.0|SolARFramework|SolARBuild@github|https://github.com/SolarFramework/SolarFramework/releases/download|shared|
```
Comment (jim-bcom) [inline packagedependencies.txt]: Is this file OK, isn't it a side-effect we discussed some time ago about the `packagedependencies.txt` file being overwritten by the compilation process?

If so, don't commit this one?

File: SolARService_MappingAndRelocalizationProxy/src/RelocalizationAndMappingGrpcServiceImpl.cpp (hunk 2 above)
Comment (granger35) [reply SolARService_MappingAndRelocalizationProxy/src/RelocalizationAndMappingGrpcServiceImpl.cpp:617]: Use push_back option and recover packagedependencies.txt as it was.

File: SolARService_MappingAndRelocalizationProxy/src/SolARService_MappingAndRelocalizationProxy.cpp (hunk 3 above)
Comment (granger35) [reply SolARService_MappingAndRelocalizationProxy/src/SolARService_MappingAndRelocalizationProxy.cpp]: has been removed

File: packagedependencies.txt (hunk 4 above)
Comment (granger35) [reply packagedependencies.txt]: Done

File: SolARService_MappingAndRelocalizationProxy/src/RelocalizationAndMappingGrpcServiceImpl.cpp (hunk 1 above)
Comment (granger35) [reply SolARService_MappingAndRelocalizationProxy/src/RelocalizationAndMappingGrpcServiceImpl.cpp]: push_back() have been applied.

File: packagedependencies.txt [hunk 5]
```diff
@@ -1,7 +1,2 @@
-SolARModuleOpenCV|0.11.0|SolARModuleOpenCV|SolARBuild@github|https://github.com/SolarFramework/SolarModuleOpenCV/releases/download
-SolARModuleOpenGL|0.11.0|SolARModuleOpenGL|SolARBuild@github|https://github.com/SolarFramework/SolarModuleOpenGL/releases/download
-SolARModuleTools|0.11.0|SolARModuleTools|SolARBuild@github|https://github.com/SolarFramework/SolARModuleTools/releases/download
-SolARModuleFBOW|0.11.0|SolARModuleFBOW|SolARBuild@github|https://github.com/SolarFramework/SolARModuleFBOW/releases/download
 SolARPipelineRelocalization|0.11.0|SolARPipelineRelocalization|SolARBuild@github|https://github.com/SolarFramework/Sample-Relocalization/releases/download
```
Comment (jim-bcom) [inline packagedependencies.txt:5]: Other runtime dependencies removed on purpose ? (despite the comment stating that `packagedependencies.txt` is as it was)

Comment (granger35) [reply packagedependencies.txt:5]: Sorry, corrected now.

**Review by jim-bcom — APPROVED** [review]
LGTM

**Pull request merged**